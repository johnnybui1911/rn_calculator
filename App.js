import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import Screen from './src/Screen'
import reducer from './src/reducer'

const stores = createStore(reducer)

export default function App() {
  return (
    <Provider store={stores}>
      <Screen />
    </Provider>
  )
}
