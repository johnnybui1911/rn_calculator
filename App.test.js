import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import App from './App'

describe('App', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<App />)
      expect(component).toMatchSnapshot()
    })
  })
})
