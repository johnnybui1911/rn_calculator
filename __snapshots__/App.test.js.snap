// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Rendering should match to snapshot 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "justifyContent": "flex-end",
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 88,
          },
          Object {
            "fontSize": 88,
          },
        ]
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "paddingLeft": 6,
          "paddingVertical": 6,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            7
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            8
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            9
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            4
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            5
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            6
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            1
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            2
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            3
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            C
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            0
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 6,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            .
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 6,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 0,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            /
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 0,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            *
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 0,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            +
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 0,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            -
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d9d9d9",
              "borderRadius": 6,
              "flex": 1,
              "justifyContent": "center",
              "margin": 6,
              "marginLeft": 0,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            =
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
